WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersPushPromptViewController.swift

```diff
@@ -0,0 +1,248 @@
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = Metrics.stackSpacing
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.distribution = .equalSpacing
+        return stackView
+    }()
+
+    private let imageView: UIImageView = {
+        let imageView = UIImageView(image: UIImage(named: Images.bellImageName))
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.tintColor = .systemYellow
+        return imageView
+    }()
+
+    private let titleLabel: UILabel = {
+        let label = UILabel()
+        label.font = WPStyleGuide.serifFontForTextStyle(.title1, fontWeight: .semibold)
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.text = TextContent.title
+        return label
+    }()
+
+    private let promptLabel: UILabel = {
+        let label = UILabel()
+        label.font = .preferredFont(forTextStyle: .body)
+        label.text = TextContent.prompt
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.textColor = .secondaryLabel
+        return label
+    }()
+
+    private let hintLabel: UILabel = {
+        let label = UILabel()
+        label.font = .preferredFont(forTextStyle: .body)
+        label.text = TextContent.hint
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.textColor = .secondaryLabel
+        return label
+    }()
+
+    private let turnOnNotificationsButton: UIButton = {
+        let button = FancyButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.isPrimary = true
+        button.setTitle(TextContent.turnOnButtonTitle, for: .normal)
+        button.addTarget(self, action: #selector(turnOnButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private let dismissButton: UIButton = {
+        let button = UIButton(type: .custom)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setImage(.gridicon(.cross), for: .normal)
+        button.tintColor = .secondaryLabel
+        button.addTarget(self, action: #selector(dismissTapped), for: .touchUpInside)
+        return button
+    }()
+
+    // MARK: - Properties
+
+    /// Indicates whether push notifications have been disabled or not.
+    ///
+    private var pushNotificationsAuthorized: UNAuthorizationStatus = .notDetermined {
+        didSet {
+            navigateIfNecessary()
+        }
+    }
+
+    /// Analytics tracker
+    ///
+    let tracker: BloggingRemindersTracker
+
+    // MARK: - Initializers
+
+    init(tracker: BloggingRemindersTracker) {
+        self.tracker = tracker
+
+        super.init(nibName: nil, bundle: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(applicationBecameActive), name: UIApplication.didBecomeActiveNotification, object: nil)
```

Comment:
can we do

```
NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification, object: nil, queue: nil) { [weak self] _ in
            self?.refreshPushAuthorizationStatus()
 }
```

This would not require the called method to be exposed to Objective C and we can avoid adding one method `applicationBecameActive()` to wrap another one.

Reply:
Thanks for the suggestion! I went with the selector-based observer because they don't need to be manually unregistered unlike block-based ones. If I do block-based, we need a variable to keep hold of the token, and to implement `deinit` to unregister the block later. (see [this blog](https://oleb.net/blog/2018/01/notificationcenter-removeobserver/) and the [Apple docs](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver) for more info).